### ISSUE TITLE
suppress events for lookup for wordlist export

### ISSUE THREAD

**Issue by balmas** [alpheios-project/alpheios-core#291]
we should probably suppress UI Events when we are doing a lookup for the wordlist export

**Comment by balmas** [comment]
@kirlat can you take a look at this one? I think it probably needs to be done before the release because I'm a little worried about the side effects of the wordlist lookups. Thanks!

**Comment by kirlat** [comment]
Sure, will work on it

**Comment by kirlat** [comment]
It has been merged to master now

**Comment by balmas** [comment]
Alpheios Components 3.3.0-qa.20200513442

**Comment by balmas** [comment]
@monzug a test scenario for this is the following:

(1) login 
(2) lookup a bunch of words so that they get added to your wordlist 
(3) logout 
(4) clear your browser cache
(5) login again 
(6) do a lookup of some other word (i.e. not one on your wordlist, maybe in a different language)
(7) export some items from your wordlist in flashcard format
(8) make sure nothing weird is happening in the UI as a result of your export -- i.e. state should be the same as after the lookup in step 6

Let me know if you have any questions about this. Thanks!

**Comment by monzug** [comment]
well there are few things that do not look ok.

1) Latin library in Chrome: the checkbox for flashcards or with filtering cannot be checked. however it works in Loeb Classics.

2) the downloaded file with flashcards option is empty. There is only word definition in it. I tried with latin (more than 25 words), then I tried with Persian (3 words). 

3) when I downloaded the Persian file with the flashcard option, for same reason it did a lookup of the last Greek word from the above word list. see screenshot

![weird-scenario](https://user-images.githubusercontent.com/41396793/82055288-42e44500-968e-11ea-9605-d07fd33a8709.JPG)

**Comment by monzug** [comment]
2) could it be because I haven't updated the env-webext.js yet?  I read your reccomendation in #200 after testing this one

**Comment by monzug** [comment]
as per 2) I was able to download the file with the flashcard option. however if I chose to limit to 1 the most  often list (of two words), the downloaded file listed both words.

as per 1) - I think it's the first language (it can be Greek or Latin) on the wordlist that does not work.

**Comment by irina060981** [comment]
@kirlat and @balmas , I tested a simple scenario - when we have data only in remote source and don't have data in IndexedDB - and homonyms are not updated without `HOMONYM_READY`
and finally - we have an empty wordlist downloaded

This happens because we send textSelector to lexis module (without any link to wordItem),
and then wordListController updates worditems from the same events:

```
export default class WordlistController {
  /**
   * @constructor
   * @param {String[]} availableLangs language codes
   * @param {PSEvent[]} events events that the controller can subscribe to
   */
  constructor (availableLangs, events) {
    this.wordLists = {}
    this.availableLangs = availableLangs
    events.TEXT_QUOTE_SELECTOR_RECEIVED.sub(this.onTextQuoteSelectorReceived.bind(this))
    events.HOMONYM_READY.sub(this.onHomonymReady.bind(this))
    events.SHORT_DEFS_READY.sub(this.onDefinitionsReady.bind(this))
    events.FULL_DEFS_READY.sub(this.onDefinitionsReady.bind(this))
    events.LEMMA_TRANSL_READY.sub(this.onLemmaTranslationsReady.bind(this))
  }
```

This way we couldn't suppress events completely for updating homonym inside wordlist

**Comment by irina060981** [comment]
>Latin library in Chrome: the checkbox for flashcards or with filtering cannot be checked. however it works in Loeb Classics. I think it's the first language (it can be Greek or Latin) on the wordlist that does not work. it does not matter which site I am in.

@monzug , I couldn't reproduce the non-availability of checkboxes inside download confirmation

- I logged in
- i have ready latin list and looked for one greek word

![image](https://user-images.githubusercontent.com/12377640/82209009-e748de80-994f-11ea-961f-caea939ef0f2.png)

Both checkbox are available and could be checked
I use the following link
http://www.thelatinlibrary.com/petronius1.html

**Comment by balmas** [comment]
> @kirlat and @balmas , I tested a simple scenario - when we have data only in remote source and don't have data in IndexedDB - and homonyms are not updated without `HOMONYM_READY`
> and finally - we have an empty wordlist downloaded
> 
> This happens because we send textSelector to lexis module (without any link to wordItem),
> and then wordListController updates worditems from the same events:
> 
> ```
> export default class WordlistController {
>   /**
>    * @constructor
>    * @param {String[]} availableLangs language codes
>    * @param {PSEvent[]} events events that the controller can subscribe to
>    */
>   constructor (availableLangs, events) {
>     this.wordLists = {}
>     this.availableLangs = availableLangs
>     events.TEXT_QUOTE_SELECTOR_RECEIVED.sub(this.onTextQuoteSelectorReceived.bind(this))
>     events.HOMONYM_READY.sub(this.onHomonymReady.bind(this))
>     events.SHORT_DEFS_READY.sub(this.onDefinitionsReady.bind(this))
>     events.FULL_DEFS_READY.sub(this.onDefinitionsReady.bind(this))
>     events.LEMMA_TRANSL_READY.sub(this.onLemmaTranslationsReady.bind(this))
>   }
> ```
> 
> This way we couldn't suppress events completely for updating homonym inside wordlist

Ah that's a good point. I'm sorry for not catching this.

**Comment by irina060981** [comment]
>I was able to download a file with the flashcard option. however if I chose to limit to 1 the most often list (of two words), the downloaded file listed both words.

@monzug , did you check both checkbox - for using filter and format as flashcard?
If I check box checkboxes in my test - it is working well

**Comment by irina060981** [comment]
@balmas and @kirlat , also we have now problems with catching the moment when short definitions are ready
this code
```
await this.lexis.lookupForWordlist(textSelector)
```
does not end when short definitions are ready this way - it downloads an empty list because short definitions are not ready

**Comment by balmas** [comment]
ok, clearly my idea to suppress the events was not a good one....

**Comment by monzug** [comment]
@irina060981 you are right! I thought that filtering was not related to the limit applied.

**Comment by monzug** [comment]
![checkbox-not checked](https://user-images.githubusercontent.com/41396793/82214461-dd899180-98e3-11ea-93b1-e8e4873be279.JPG)

**Comment by irina060981** [comment]
>how to repro (at least in my environment). go to the already opened latinlibrary or loeb, not log in, wordlist is empty, double click on few words, go to wordlist to download the list. the checkboxes for filtering and flashcard cannot be checked.
note that it's the first thing I do on these sites (in my case, they have been sitting there for few hours). if I log in or browse inflection or others and then go to wordlist, then it works fine.

I was not able to reproduce this bug and I even couldn't imagine what could be the source of the situation, when checkbox is not checked on click here

@balmas and @kirlat , may be some of you could reproduce this ?

**Comment by balmas** [comment]
per our discussion earlier this week, we'll reinstate the events and suppress only the ui messages

**Comment by irina060981** [comment]
Merged and closed

**Comment by balmas** [comment]
Alpheios Components 3.3.0-qa.20200522457

**Comment by monzug** [comment]
I did follow Bridget's scenario just as a start for testing. well, all went smoothly except when downloading a file
for latin, I select to limit to this session and it downloaded the 4 words for current session. However for greek, I did not specify a limit and I checked to download with filtering and with flashcard. only one word was saved in the downloaded file, but 25 words from the Greek wordlist were marked as current session (24 are latin words, the only one that get downloaded is a Greek word).

**Comment by balmas** [comment]
> 1. Is it ok that a greek wordlist only downloads greek words? or should all the first 25 words in the list be downloaded no matter which language they are?

I think the behavior is correct - the limit should apply per language -- i.e. so if only one greek word has been looked up in a session, when downloading the greek list, if the filter is to current session it should contain only 1 word

**Comment by balmas** [comment]
> but 25 words from the Greek wordlist were marked as current session (24 are latin words, the only one that get downloaded is a Greek word from wordlist). I did also try without filtering and I got the same results. .... the words that are 'downloaded' (or better, the first 25 words in the wordlist) are marked as current session while they are not.

I'm a little confused about this - what are the 24 words? Are they Latin words that are appearing on the Greek list? I.e. Latin words looked up while page language was set to Greek?

**Comment by monzug** [comment]
yes. 24 are latin words in a greek words list

**Comment by balmas** [comment]
>some words are repeated like fortuna, primus, based on the part of speech. the 'missing' words are the one with a failed lexical query. This should be ok. do you think we should say to expect only the words with lexical data, maybe in 'Note: the Flashcard export is limited to a maximum of 25 words'?

I agree the behavior is ok, but I'm not sure if it warrants a note on the download page -- do you think we could just make a note in the FAQ?

**Comment by balmas** [comment]
> yes. 24 are latin words in a greek words list

Ok, so they aren't getting downloaded because they have no data, right? I think this is okay, it should be an edge case.

**Comment by balmas** [comment]
>first it takes quite sometimes to create/download the file. the window shrinks and expands minimally few times. see screenshot of the loading time

I don't know if we can do anything about this. I'm hoping this is more of an edge case -- i.e ideally the words should be in the browser cache and not needing to be re-retrieved.

**Comment by balmas** [comment]
>if I double-click on a word in wordlist with no lexical data, it loads for ever.
This needs to be fixed via #353

**Comment by balmas** [comment]
@monzug we can verify this one right?
